src/main/java/org/asciidoctor/asciidoclet/AsciiDocTrees.java

```diff
@@ -151,7 +153,12 @@ public DocTreePath getDocTreePath(FileObject fileObject, PackageElement packageE
     public Element getElement(DocTreePath path) {
         return docTrees.getElement(path);
     }
-
+    
+//    @Override
+    public TypeMirror getType(DocTreePath path) {
+        return null;
+    }
```

Comment:
Commented out `@Override`.
This made the source code compilable with Java11, Java17, and Java21.
Also, the rendered docs looked fine.

Not sure with the compiled binary (for instance compiled by Java11) works for all of them.

Reply:
I made sure that the compiled binary can render Java21-based project.
https://github.com/dakusui/doclet-exercise